internal/fs/hns_bucket_test.go

```diff
@@ -100,3 +100,101 @@ func (t *HNSBucketTests) TestDeleteFolder() {
 	_, err = os.Stat(dirPath)
 	assert.NotNil(t.T(), err)
 }
+
+func (t *HNSBucketTests) TestRenameFolderWithSrcDirectoryDoesNotExist() {
+	oldDirPath := path.Join(mntDir, "foo_not_exist")
+	newDirPath := path.Join(mntDir, "foo_rename")
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NotNil(t.T(), err)
+	_, err = os.Stat(newDirPath)
+	assert.NotNil(t.T(), err)
+}
+
+func (t *HNSBucketTests) TestRenameFolderWithDstDirectoryIsNotEmpty() {
+	oldDirPath := path.Join(mntDir, "foo")
+	_, err = os.Stat(oldDirPath)
+	assert.NoError(t.T(), err)
+	newDirPath := path.Join(mntDir, "bar")
+	_, err = os.Stat(newDirPath)
+	assert.NoError(t.T(), err)
+
+	err = os.Rename(oldDirPath, newDirPath)
```

Comment:
We haven't really put any file in the dst directory, so either the test name is wrong, or you missed to add a file in the dst directory.

Reply:
The test is

A/a.txt
B/b.txt

We are renaming A to B and it should throw an issue.

here we are renaming `foo` to `bar`.  and  `bar` is not empty 

> We haven't really put any file in the dst directory, so either the test name is wrong, or you missed to add a file in the dst directory.

We have done it in SetUp.

Reply:
can you add a comment about this.